source/guides/clear/swupd-3rd-party.rst

```diff
@@ -0,0 +1,273 @@
+.. _swupd-3rd-party:
+
+swupd 3rd-party 
+###############
+
+Upstream |CL| offers a plethora of `bundles`_ to choose from.  
+
+For users who want access to additional software outside of the distro, 
+|CL| provides support for 3rd-party bundles.  
+
+There are two components to 3rd-party bundles:
+
+* Use :command:`mixer` to create 3rd-party bundles
+
+* Use the :command:`swupd` subcommand :command:`3rd-party` to manage repos, 
+  consume and manage bundles
+
+Follow this guide to set up a web server to host 3rd-party bundles, 
+build an example 3rd-party bundle, install and manage the bundle on a 
+client system. 
+
+.. contents::
+   :local:
+   :depth: 1
+
+Prerequisite
+*************
+
+* Familiarity with :command:`mixer`
+* Familiarity with :command:`swupd`
+* You must be running |CL| version 32570 or higher
+
+.. include:: ./mixer.rst
+   :start-after: set-up-nginx-web-server-start:
+   :end-before: set-up-nginx-web-server-end:
+
+Create directory to hold 3rd-party app
+**************************************
+
+#. Create a top-level directory to hold all apps.
+	
+   .. code-block:: bash
+      
+      mkdir ~/my-3rd-party-apps && pushd $_
+
+#. Create a directory for each app and put the content of your software in it.  
+
+   In this example, the `helloclear.sh` app simply prints
+   "Hello Clear!" when invoked.  
+
+   .. code-block:: bash
+
+      # make helloclear directory
+      mkdir -p helloclear/usr/bin && pushd $_
+
+      # create helloclear.sh script
+      cat > helloclear.sh << EOF
+      #!/bin/bash
+      echo "Hello Clear!"
+      EOF
+
+      # make script executable
+      chmod +x helloclear.sh
+
+      popd
+
+   .. note:: 
+      
+      * You can put whatever you want in your app's directory. All of the content 
+        within each directory will get copied onto the client system under 
+        :file:`/opt/3rd-party/<app-directory>`.
```

Comment:
In the swupd-client side, I believe it gets copied to `/opt/3rd-party/bundles/<repo_name>`. @castulo Can you please confirm.

Also, please note that, in mixer, the `<top-level-dir>` specified using the `contents(<top-level-dir>)` keyword does not get copied. Its only the dirs/files within the `<top-level-dir>` that gets copied.

Reply:
got it.